src/actions/transactionEstimateActions.js

```diff
@@ -146,8 +149,12 @@ export const estimateTransactionsAction = (
     const supportedAssetsPerChain = supportedAssetsPerChainSelector(getState());
     const chainSupportedAssets = supportedAssetsPerChain[chain] ?? [];
 
+    const gasTokenAddress = preferredGasTokenSelector(getState()) === PLR
```

Comment:
Seems like all non PLR values lead to using native asset. Is that intened?

Reply:
Q2: is PLR availalbe on all chains? maybe we need to default to nativeAsset in case there is no PLR for given side chain

Reply:
Yes, that is intended at this moment in time. PLR is available on all chains except xDai. If there's no or empty PLR address returned by `getPlrAddressForChain` then `findAssetByAddress` would return no asset therefore gas token will be native asset.

Actually now thinking that this should be replaced into:

```
const gasTokenAddress = preferredGasTokenSelector(getState()) === PLR
      ? getPlrAddressForChain(chain)
      : null
const gasToken = useGasToken && gasTokenAddress
      ? findAssetByAddress(chainSupportedAssets, gasTokenAddress)
      : null;
```